File: sample-domain/npm/src/aggregates/parent/child.ts
import { EntityBase, IEntity } from "../../abstractions/entity";

export interface IChild extends IEntity {
    parentId: number;
    description: string;
}

export class Child extends EntityBase implements IChild {
    parentId: number;
    description: string;
    constructor(parentId: number, description: string) {
        super();
        this.parentId = parentId;
        this.description = description;
    }
    hydrate(state: Partial<IChild>): Child {
        this.id = state.id || 0;
        this.parentId = state.parentId || 0;
        this.description = state.description || "";
        return this;
    }
}

File: sample-domain/dotnet/src/aggregates/parent/child.cs
using Sample.Domain.Abstractions;

namespace Sample.Domain.Aggregates.Parent
{
    public class Child : EntityBase, IEntity
    {
        public virtual int ParentId { get; set; }
        public virtual string Description { get; set; }
        private Child() { }
        public Child(int parentId, string description) 
        {
            ParentId = parentId;
            Description = description;
        }
    }
}